### DISCUSSION TITLE
own implementation of generator

### DISCUSSION THREAD

**Discussion by McP4nth3r** [QuestPDF/QuestPDF#1007] [General]
Is there a way to program the whole generation with QuestPDF not in fluent and make all checks of space by our own to dynamicly generate complex designs? the reason is because the dynamic component is for our use case not enough. I would be nice to overload the default Generator to build an own implementation.

**Comment by girlpunk** [comment]
I don't believe this is currently supported by QuestPDF, however are you able to share a small example of how you would intend to use the library without the fluent API? I'm a little unsure on your intended use case

**Reply by McP4nth3r** [reply]
i would like to build the document step by step like the dynamic element.

Create Element save it in a reference object. check the sizes and add it to the document and this step by step to check if a text is fit on a page or need to be truncated and the rest displayed on another page but with a dynamic table with subtotals and more.

**Reply by BeneHenke** [reply]
You can build the document step by step if you save things like the page/ column/ text in a variable and then add on to them if you need to.

**Reply by McP4nth3r** [reply]
Ok with dynamic component or how you mean? The fluent API expose only voids and I can't save them in a variable.

**Reply by BeneHenke** [reply]
When creating the column like this:
```
ColumnDescriptor columnDesc;
.Column(column =>
{
    column.Item().Background(Colors.Grey.Medium).Height(50);
    column.Item().Background(Colors.Grey.Lighten1).Height(100);
    column.Item().Background(Colors.Grey.Lighten2).Height(150);
    columnDesc = column;
});
```
You can save the columnDescriptor and access when you need to.

**Reply by McP4nth3r** [reply]
I will try it when I'm home thanks for the tip this could work 👌🏻

**Reply by McP4nth3r** [reply]
i can't get the Sizes of the generated document (page). The descriptors only exposes voids and setters not getters. Do you have an example on how to create a text => check the size and shrink the text if height is greater than 50 milimeters

**Reply by BeneHenke** [reply]
Sorry if my previous answer wasn't helping. Have you tried this https://www.questpdf.com/api-reference/scale-to-fit.html ? If it doesn't work you would probably need the dynamic component again.

**Reply by McP4nth3r** [reply]
yeah, the dynamic component is required in my case. But the problem still exists. I can't truncate the text and display rest on the next page as continues content, because the compose method of dynamic component is executed more times per page.

**Reply by girlpunk** [reply]
The dynamic component being called multiple times is expected behaviour, as QuestPDF will render components multiple times in different sizes, in attempts to make the content fit with the various constraints specified. Are you able to share a little more about your use case, and why the dynamic component is required? If making text wrap onto the next page is your only need, this is handled natively in QuestPDF by the Text element.

**Reply by McP4nth3r** [reply]
I will made a example solution asap 👌
